source/common/stats/thread_local_store.cc

```diff
@@ -329,12 +328,62 @@ StatType& ThreadLocalStoreImpl::ScopeImpl::safeMakeStat(
   // If we have a TLS cache, insert the stat.
   if (tls_cache) {
     tls_cache->insert(std::make_pair((*central_ref)->statName(), *central_ref));
+    ENVOY_LOG_MISC(debug, "stat added to central cache map");
+    name.debugPrint();
+    ENVOY_LOG_MISC(debug, "central_cache_map = {}", static_cast<void*>(&central_cache_map));
+    ENVOY_LOG_MISC(debug, "this = {}", static_cast<const void*>(this));
+    ENVOY_LOG_MISC(debug, "parent_ = {}", static_cast<void*>(&parent_));
   }
 
   // Finally we return the reference.
   return **central_ref;
 }
 
+// TODO(ahedberg): This is largely duplicated from safeMakeStat and should
+// be cleaned up at some point.
+template <class StatType>
+const StatType* ThreadLocalStoreImpl::ScopeImpl::safeGetStat(
+    StatName name, StatMap<std::shared_ptr<StatType>>& central_cache_map,
+    StatMap<std::shared_ptr<StatType>>* tls_cache, StatNameHashSet* tls_rejected_stats) const {
+
+  if (tls_rejected_stats != nullptr &&
```

Comment:
I think we don't need to consult the rejected stats  lookup here -- it won't be in the scope-map if it was rejected.

Reply:
Removed.